src/directive/translate.js

```diff
@@ -303,7 +302,9 @@ function translateDirective($translate, $q, $interpolate, $compile, $parse, $roo
         if (translateValuesExist || translateValueExist || iAttr.translateDefault) {
           scope.$watch('interpolateParams', updateTranslations, true);
         }
-        scope.$watch('translateLanguage', updateTranslations);
+
+        // Replaced watcher on translateLanguage with event listener
+        var unbindTranslateLanguage = scope.$on('translateLanguageChanged', updateTranslations);
```

Comment:
Regarding the watcher reduction: with each isolated scope, the previous version would register one 'translateLanguage' watcher regardless the presence of the translate-language directive usage. The commit contains a new demo file: ex14_translate_language.htm that uses twice the translate-language directive (previous implementation: 14 watchers, current implementation: 12 watchers) by avoiding adding 'translateLanguage' watchers for the following DOM node elements:
<h2 translate="SUBHEADER">Subheader</h2>
<p translate>HTML_KEYS</p>